guides/micronaut-openapi-generator-server/micronaut-openapi-generator-server.adoc

```diff
@@ -0,0 +1,497 @@
+
+NOTE: You can read more about writing schemas in the
+link:https://swagger.io/docs/specification/data-models/["Data Models (Schemas)" OpenAPI guide].
+
+Save the file and proceed to the next part of the guide.
+//Altogether it should look like this: link:@sourceDir@/library-definition.yaml[library-definition.yaml].
+
+== Generating server API from template
+
+Now we will generate server API files from our definition.
+Open the terminal in the same folder as `libaray-definition.yaml` file and run the following command:
+
+[source,bash]
+----
+java -jar openapi-generator-cli-5.4.0.jar generate \
+    -g java-micronaut-server \# <1>
+    -i library-definition.yaml \# <2>
+    -o ./ \# <3>
+    -p controllerPackage=example.micronaut.library.controller \# <4>
+    -p modelPackage=example.micronaut.library.model \# <5>
+    -p build=@build@ \# <6>
+    -p test=junit# <7>
+----
+<1> Specify that we will use java Micronaut server generator.
+<2> Specify our OpenAPI definition file as `library-definition.yaml` which we just created.
+<3> Specify the output directory to be the current directory (`./`).
+You can specify it to be a different one if you want (e.g. `library-server`).
+<4> We provide generator-specific properties starting with `-p`.
+We want all the controllers to be generated in the `example.micronaut.library.controller` package.
+<5> We want all the models (data models, like `BookInfo`) to be in `example.micronaut.library.model` package.
+<6> We want to use @build@ as build tool. The supported values are `gradle`, `maven` and `all`.
+If nothing is specified, both maven and gradle files are generated.
+<7> We want to use JUnit 5 for testing. The supported values are `junit` (JUnit 5) and `spock`.
+If nothing is specified, `junit` is used by default.
+
+[WARNING]
+=========
+If you are using Windows command prompt, run:
+=========
+[source,bash]
+----
+java -jar openapi-generator-cli-5.4.0.jar generate -g java-micronaut-server -i library-definition.yaml -o ./ -p controllerPackage=example.micronaut.library.controller -p modelPackage=example.micronaut.library.model -p build=@build@ -p test=junit
+----
+
+[NOTE]
+=========
+If you want to view all the available parameters for micronaut server generator, run
+[source,bash]
+----
+java -jar openapi-generator-cli-5.4.0.jar config-help \
+    -g java-micronaut-server
+----
+=========
+
+[NOTE]
+=========
+If you plan to change the definition file and regenerate files, consider setting the `-p generateControllerAsAbstract=true`
+parameter (We don't recommend doing it during this guide, though). In this case an abstract class will be generated
+for the API, while all the logic needs to be implemented in a different class (that inherits the API abstract class).
+This way your changes won't be overwritten by generation, but API will be updated.
+=========
+
+After running OpenAPI generator CLI should output information about generated files.
```

Comment:
"generator CLI" -> "generator, the CLI"

Reply:
How about "After running, the OpenAPI generator CLI will output information about generated files."